File: speakers.py
#/usr/bin/env python

import yaml
import twitter
import urllib

config = yaml.safe_load(file('twitter-secrets.yml', 'r'))

twitterApi = twitter.Api(consumer_key=config['consumer_key'],
                         consumer_secret=config['consumer_secret'],
                         access_token_key=config['access_token_key'],
                         access_token_secret=config['access_token_secret'])

data = yaml.safe_load(file('_data/speakers.yml', 'r'))

for speaker in data:
    if speaker['twitter']:
        screen_name = speaker['twitter']
        print 'Fetching data for "%s"' % screen_name
        user = twitterApi.GetUser(screen_name=screen_name)
        url = user.GetProfileImageUrl()
        filename = 'images/speakers/' + screen_name + '-' + url.split('/')[-1]
        urllib.urlretrieve(url, filename)

        speaker['image'] = '/' + filename
        speaker['about'] = user.GetDescription()

output = file('_data/speakers.yml', 'w')
yaml.safe_dump(data, output, encoding='utf-8', indent=2, allow_unicode=True)

File: speakers.php
#!/usr/bin/env php
<?php

require __DIR__ . '/vendor/autoload.php';
require __DIR__ . '/twitter-secrets.php';

use Symfony\Component\Yaml\Yaml;

function replaceShortUrls($text, $urls = [])
{
	foreach ($urls as $url) {
		$link = '<a href="' . $url->expanded_url . '" target="_blank">' . $url->display_url . '</a>';
		$text = str_replace($url->url, $link, $text);
	}
	return $text;
}

$twitter = new Twitter($consumerKey, $consumerSecret, $accessToken, $accessTokenSecret);

$speakersFile = __DIR__ . '/app/_data/speakers.yml';
$speakers = Yaml::parse(file_get_contents($speakersFile));

foreach ($speakers as $key => $speaker) {
	if (isset($speaker['twitter'])) {
		$username = $speaker['twitter'];
		echo 'Updating ' . $username . "'s profile ...\n";
		$info = $twitter->loadUserInfo($username);
		$profileImage = file_get_contents($info->profile_image_url);
		$pathinfo = pathinfo($info->profile_image_url);
		$newProfileImageName = '/images/speakers/' . $username . '-' . $pathinfo['basename'];
		file_put_contents('./app' . $newProfileImageName, $profileImage);
		$speakers[$key]['image'] = $newProfileImageName;
		if (isset($info->description)) {
			$speakers[$key]['about'] = replaceShortUrls($info->description, $info->entities->description->urls);
		}
		if (isset($info->url)) {
			$speakers[$key]['url'] = replaceShortUrls($info->url, $info->entities->url->urls);
		}
	}
}

usort($speakers, function ($a, $b) {
	return strcmp($a['_id'], $b['_id']);
});

file_put_contents($speakersFile, Yaml::dump($speakers));
